omegaconf/base.py

```diff
@@ -429,72 +424,61 @@ def _resolve_interpolation_from_parse_tree(
     def _resolve_node_interpolation(
         self,
         inter_key: str,
-        throw_on_missing: bool,
-        throw_on_resolution_failure: bool,
-    ) -> Optional["Node"]:
+    ) -> "Node":
         """A node interpolation is of the form `${foo.bar}`"""
-        root_node, inter_key = self._resolve_key_and_root(inter_key)
-        parent, last_key, value = root_node._select_impl(
-            inter_key,
-            throw_on_missing=throw_on_missing,
-            throw_on_resolution_failure=throw_on_resolution_failure,
-        )
+        try:
+            root_node, inter_key = self._resolve_key_and_root(inter_key)
+        except ConfigKeyError as exc:
+            raise InterpolationKeyError(str(exc))
+
+        try:
+            parent, last_key, value = root_node._select_impl(
+                inter_key,
+                throw_on_missing=True,
+                throw_on_resolution_failure=True,
+            )
+        except MissingMandatoryValue as exc:
+            raise InterpolationToMissingValueError(str(exc))
+        except ConfigKeyError as exc:
+            raise InterpolationKeyError(str(exc))
```

Comment:
here too.
Also: can you make sure all those different error cases are covered in test_errors.py?

Reply:
Also done in 5718bf0